traits/observation/tests/test_observe.py

```diff
@@ -653,3 +655,74 @@ def test_apply_observers_different_target(self):
         # then
         # the handler should be called twice as the targets are different.
         self.assertEqual(handler.call_count, 2)
+
+
+# ---- Low-level tests for async dispatch_same ------------------------------
+
+
+class TestAsyncDispatchSame(unittest.IsolatedAsyncioTestCase):
+    """Test low-level async dispatch."""
+
+    def setUp(self):
+        from traits.observation.observe import _active_handler_tasks
+
+        # ensure no lingering references to handler tasks after test run
+        self.addCleanup(_active_handler_tasks.clear)
```

Comment:
This suggests that we might want a public function to clear references to active tasks - I could see it being useful at application teardown time. That could wait for another PR, though. (We might also want a way to do a synchronous wait or wait-with-timeout for active tasks to complete.)

Reply:
I had that public function in my initial draft, but wanted this PR to be the minimal needed.  It's very easy to add if we need it.